cicecore/cicedyn/infrastructure/io/io_pio2/ice_history_write.F90

```diff
@@ -192,73 +191,106 @@ subroutine ice_write_hist (ns)
+         call ice_pio_check(pio_put_att(File,varid,'calendar','360_day'), &
+              subname//' ERROR: defining att calendar1',file=__FILE__,line=__LINE__)
+      elseif (days_per_year == 365 .and. .not.use_leap_years ) then
+         call ice_pio_check(pio_put_att(File,varid,'calendar','noleap'), &
+              subname//' ERROR: defining att calendar2',file=__FILE__,line=__LINE__)
+      elseif (use_leap_years) then
+         call ice_pio_check(pio_put_att(File,varid,'calendar','Gregorian'), &
+              subname//' ERROR: defining att calendar3',file=__FILE__,line=__LINE__)
+      else
+         call abort_ice(subname//' ERROR: invalid calendar settings')
+      endif
 
-        if (hist_avg(ns) .and. .not. write_ic) then
-          status = pio_put_att(File,varid,'bounds','time_bounds')
-        endif
+      if (hist_avg(ns) .and. .not. write_ic) then
+         call ice_pio_check(pio_put_att(File,varid,'bounds','time_bounds'), &
+              subname//' ERROR: defining att bounds time_bounds',file=__FILE__,line=__LINE__)
+      endif
 
-        ! Define attributes for time_bounds if hist_avg is true
-        if (hist_avg(ns) .and. .not. write_ic) then
-          dimid2(1) = boundid
-          dimid2(2) = timid
-          status = pio_def_var(File,'time_bounds',pio_double,dimid2,varid)
-          status = pio_put_att(File,varid,'long_name', &
-                                'time interval endpoints')
-
-          if (days_per_year == 360) then
-             status = pio_put_att(File,varid,'calendar','360_day')
-          elseif (days_per_year == 365 .and. .not.use_leap_years ) then
-             status = pio_put_att(File,varid,'calendar','noleap')
-          elseif (use_leap_years) then
-             status = pio_put_att(File,varid,'calendar','Gregorian')
-          else
-             call abort_ice(subname//'ERROR: invalid calendar settings')
-          endif
-
-          write(cdate,'(i8.8)') idate0
-          write(title,'(a,a4,a1,a2,a1,a2,a1,i2.2,a1,i2.2,a1,i2.2)') 'days since ', &
-                cdate(1:4),'-',cdate(5:6),'-',cdate(7:8),' ', &
-                hh_init,':',mm_init,':',ss_init
-          status = pio_put_att(File,varid,'units',trim(title))
-        endif
+      ! Define attributes for time_bounds if hist_avg is true
+      if (hist_avg(ns) .and. .not. write_ic) then
+         dimid2(1) = boundid
+         dimid2(2) = timid
+         call ice_pio_check(pio_def_var(File,'time_bounds',pio_double,dimid2,varid), &
+              subname//' ERROR: defining var time_bounds',file=__FILE__,line=__LINE__)
+         call ice_pio_check(pio_put_att(File,varid,'long_name', 'time interval endpoints'), &
+              subname//' ERROR: defining att long_name time interval endpoints',file=__FILE__,line=__LINE__)
+
+         if (days_per_year == 360) then
+            call ice_pio_check(pio_put_att(File,varid,'calendar','360_day'), &
+                 subname//' ERROR: defining att calendar1 tb',file=__FILE__,line=__LINE__)
```

Comment:
I'm not sure what "tb" is supposed to mean, but I would suggest the errors call out which calendar is being checked (360_day, noleap and Gregorian).